lapis/src/test/kotlin/org/genspectrum/lapis/model/AdvancedQueryFacadeTest.kt

```diff
@@ -0,0 +1,788 @@
+                    ),
+                ),
+            ).map { arrayOf(it.description, it.query, it.expected) }
+
+        @JvmStatic
+        fun invalidRangeQueryProvider() =
+            listOf(
+                InvalidTestCase(
+                    "date",
+                    "date<=2021.01.01",
+                    "'2021.01.01' is not a valid date",
+                ),
+                InvalidTestCase(
+                    "metadata",
+                    "some_metadata<=2021.01.01",
+                    "expression <= cannot be used for STRING",
+                ),
+                InvalidTestCase(
+                    "intField",
+                    "intField>=One",
+                    "'One' is not a valid int",
+                ),
+                InvalidTestCase(
+                    "floatField",
+                    "floatField>=One",
+                    "'One' is not a valid float",
+                ),
+            ).map { arrayOf(it.description, it.query, it.expected) }
+
+        @JvmStatic
+        fun invalidMetadataProvider() =
+            listOf(
+                InvalidTestCase(
+                    "dateFrom",
+                    "dateFrom=2020-01-01",
+                    "Metadata field dateFrom does not exist",
+                ),
+                InvalidTestCase(
+                    "dateTo",
+                    "dateTo=2020-01-01",
+                    "Metadata field dateTo does not exist",
+                ),
+                InvalidTestCase(
+                    "intFieldFrom",
+                    "intFieldFrom=1",
+                    "Metadata field intFieldFrom does not exist",
+                ),
+                InvalidTestCase(
+                    "intFieldTo",
+                    "intFieldTo=1",
+                    "Metadata field intFieldTo does not exist",
+                ),
+                InvalidTestCase(
+                    "floatFieldFrom",
+                    "floatFieldFrom=1",
+                    "Metadata field floatFieldFrom does not exist",
+                ),
+                InvalidTestCase(
+                    "floatFieldTo",
+                    "floatFieldTo=1",
+                    "Metadata field floatFieldTo does not exist",
+                ),
+            ).map { arrayOf(it.description, it.query, it.expected) }
+    }
+
+    @ParameterizedTest(name = "valid {0} with >= and <=")
+    @MethodSource("validRangeQueryProvider")
+    fun `test valid advanced queries`(
+        description: String,
+        query: String,
+        expected: SiloFilterExpression,
+    ) {
+        val result = underTest.map(query)
+        assertThat(result, equalTo(expected))
+    }
+
+    @ParameterizedTest(name = "invalid {0} with >= and <=")
+    @MethodSource("invalidRangeQueryProvider")
+    fun `test invalid advanced queries`(
+        description: String,
+        query: String,
+        expected: String,
+    ) {
```

Comment:
The IDE shows me a warning "Multiple parameters have to be wrapped in 'Arguments' ". We can simply remove the `map` in the provider function and pass the `InvalidTestCase` to this function
```suggestion
    @MethodSource("invalidRangeQueryProvider")
    fun `test invalid advanced queries`(
        testCase: InvalidTestCase
    ) {
```

(same for the `invalidMetadataProvider`)

Reply:
Actually we could also merge the tests. They are essentially the same.